sql/analyzer/resolve_columns.go

```diff
@@ -617,6 +717,8 @@ func indexColumns(ctx *sql.Context, a *Analyzer, n sql.Node, scope *Scope) (map[
 		shouldIndexChildNode = false
 	case *plan.RecursiveCte, *plan.Union:
 		shouldIndexChildNode = false
+	case *plan.TableAlias:
```

Comment:
Problems with `indexColumns` arise for queries like this
1. `SELECT names FROM orgs, json_table(orgs.members'$[*]' columns (names VARCHAR(100) path '$')) AS jt;` and
2. `SELECT names FROM orgs AS o, json_table(o.members'$[*]' columns (names VARCHAR(100) path '$')) AS jt;`

Initially, my change was to add the left tree to `scope` before calling `indexColumns` on the JSONTable like this:
```
rightScope := scope.NewScope(left)
columns, err := indexColumns(ctx, a, jt, rightScope)
```
Then, the schemas should be correctly indexed by `indexSchema(scope.Schema())`
However, `scope.Schema()` doesn't actually get the schema of the nodes in `scope`, but rather the schemas of all of their children.
```go
var schema sql.Schema
	for _, n := range s.OuterToInner() {
		for _, n := range n.Children() {
			if n.Resolved() {
				schema = append(schema, n.Schema()...)
				continue
			}
			...
		}
	}
	return schema
}
```

For case 1, `left` is a `*plan.ResolvedTable`, which has no children, so the schema is empty
For case 2, `left` is a `*plan.TableAlias` with a `*plan.ResolvedTable` child, so it indexes the schema of the `ResolvedTable`, skipping over logic in `TableAlias.Schema` that changes the Source of each column to the table's aliased name

Reply:
Your original plan is the right one. This query should work but fails with this PR:

```go
	{
		Name: "join table, table, json_table",
		SetUpScript: []string{
			`create table t1 (x json)`,
			`insert into tbl values ('[{"a":5},{"a":6}]')`,
			`create table t2 (y int primary key)`,
			`insert into t2 values (3), (4)`,
			`create table tbl (j json)`,
			`insert into tbl values ('[{"a":5},{"a":6}]')`,
		},
		Query: "select t1.x, t2.y, jt.a from t1, t2, tbl, json_table(t1.x, '$[*]' columns (a int path '$.a')) as jt",
		Expected: []sql.Row{
			{1, 3, 5},
			{1, 3, 6},
			{1, 4, 5},
			{1, 4, 6},
			{2, 3, 5},
			{2, 3, 6},
			{2, 4, 5},
			{2, 4, 6},
		},
	},
```

Because indexColumns isn't recursive (works only on a single node), your code only works if the referenced table is immediately to the left of the JSON_TABLE, not any more distant than that. This is because join.Left() and its children won't contain the columns you need.

You're right that scope indexes child nodes, and this is by design. There's any easy way around this though: just construct a synthetic Project node with join.Left as its parent, and use that as the scope for indexColumns.